### ISSUE TITLE
[HELP] Why my osd has 300GB RAW USE even there is no pg  on the osd

### ISSUE THREAD

**Issue by TimeLordFang** [rook/rook#14290]
**I have a Ceph Cluster(v17.2.7) with 300 nodes and 3000+ osds deployed by Rook(v1.13.1)**

Except for some metadata storage pools which using 3 size replicated , all others use erasure coding (6+2, 7+3)

And this is the df status:
```
--- RAW STORAGE ---
CLASS     SIZE    AVAIL     USED  RAW USED  %RAW USED
hdd     25 PiB   23 PiB  2.4 PiB   2.4 PiB       9.29
ssd    152 TiB  110 TiB   42 TiB    42 TiB      27.68
TOTAL   26 PiB   23 PiB  2.4 PiB   2.4 PiB       9.39
```
```bash 
bash-4.4$ ceph osd df > /tmp/osddf0601.txt; head /tmp/osddf0601.txt -n 1;grep ssd /tmp/osddf0601.txt  |grep -E "447|1.7 TiB"
ID    CLASS  WEIGHT   REWEIGHT  SIZE     RAW USE  DATA     OMAP      META     AVAIL    %USE   VAR   PGS  STATUS
 598    ssd  1.74619   1.00000  1.7 TiB  387 GiB  384 GiB   108 MiB  3.4 GiB  1.4 TiB  21.65  2.30   13      up
 310    ssd  1.74619   1.00000  1.7 TiB  382 GiB  379 GiB   101 MiB  3.1 GiB  1.4 TiB  21.37  2.27   12      up
 779    ssd  1.74619   1.00000  1.7 TiB  385 GiB  383 GiB    20 KiB  2.4 GiB  1.4 TiB  21.54  2.29   12      up
 798    ssd  0.43669   1.00000  447 GiB  377 GiB  375 GiB   100 MiB  2.2 GiB   70 GiB  84.37  8.98    2      up
1967    ssd  1.74619   1.00000  1.7 TiB  384 GiB  378 GiB   2.2 GiB  3.7 GiB  1.4 TiB  21.46  2.28    9      up
2025    ssd  0.43669   1.00000  447 GiB  380 GiB  378 GiB   271 KiB  2.4 GiB   67 GiB  85.07  9.05    5      up
1239    ssd  0.43669   1.00000  447 GiB  374 GiB  373 GiB     1 KiB  1.2 GiB   73 GiB  83.62  8.90    0      up
1564    ssd  1.74619   1.00000  1.7 TiB  384 GiB  381 GiB     1 KiB  2.4 GiB  1.4 TiB  21.46  2.28    8      up
2644    ssd  0.43669   1.00000  447 GiB  377 GiB  374 GiB    66 KiB  2.6 GiB   71 GiB  84.21  8.96    1      up
1245    ssd  1.74619   1.00000  1.7 TiB  379 GiB  377 GiB    12 KiB  2.4 GiB  1.4 TiB  21.19  2.25    6      up
 275    ssd  1.74619   1.00000  1.7 TiB  392 GiB  386 GiB   1.7 GiB  4.0 GiB  1.4 TiB  21.91  2.33   16      up
 276    ssd  0.43669   1.00000  447 GiB  374 GiB  373 GiB     1 KiB  933 MiB   74 GiB  83.56  8.89    0      up
```

### Here are my Problems

1. The actual usage of all storage pools does not exceed 1PB, but the usage statistics of OSD exceed 2PB. 
2. Why the ssd osds with  different WEIGHT have same RAW USE , and some even have no pg

### What I have already done：
1. ceph tell osd.* compact
2. purge deleted rdb image
3. radosgw-admin gc process --include-all

**Can anyone help me or give me some suggestions.
Are there any parameters that I did not set correctly?**

**Comment by travisn** [comment]
What do the [Ceph balancer setttings](https://docs.ceph.com/en/reef/rados/operations/balancer/) show? It seems the balancer is not balancing the PGs as expected.

**Comment by TimeLordFang** [comment]
> What do the [Ceph balancer setttings](https://docs.ceph.com/en/reef/rados/operations/balancer/) show? It seems the balancer is not balancing the PGs as expected.

```bash
bash-4.4$ ceph balancer eval
current cluster score 0.067263 (lower is better)
bash-4.4$ ceph balancer pool ls
bash-4.4$ ceph config get  mgr mgr/balancer/pool_ids

bash-4.4$ ceph balancer status
{
    "active": true,
    "last_optimize_duration": "0:00:03.751798",
    "last_optimize_started": "Tue Jun  4 01:41:30 2024",
    "mode": "upmap",
    "no_optimization_needed": true,
    "optimize_result": "Unable to find further optimization, or pool(s) pg_num is decreasing, or distribution is already perfect",
    "plans": []
}
bash-4.4$ ceph balancer eval
current cluster score 0.067262 (lower is better)
bash-4.4$
```


according to the [ceph docs](https://docs.ceph.com/en/reef/rados/operations/balancer/#throttling) 
 > Automatic balancing can be restricted to certain pools. By default, the value of this setting is an empty string, so that all pools are automatically balanced

I think my ceph balancer settings is ok, or any other configuration I missed?

Even if the balancer does not work properly , Why tho osd with 0 pg has 300GB RAW USE?

**Comment by travisn** [comment]
What is the total PG count for pools on the `ssd` device class? The PG count per OSD seems very low. You might need to increase the PG could for these `ssd` pools so the PGs will be more evenly spread. 

Not sure about the raw usage, agreed it doesn't make sense especially where there are 0 PGs. I would recommend emailing the Ceph list or ask in Ceph slack since these are core Ceph issues, and Rook does not control them.

**Comment by TimeLordFang** [comment]
@travisn 
> What is the total PG count for pools on the `ssd` device class? The PG count per OSD seems very low. You might need to increase the PG could for these `ssd` pools so the PGs will be more evenly spread.

Oh that's my another question. At first I increased the PG count for some pools(adout 30~50 PGs per OSD).but the autoscale_mode was turned  on by default. It just decreased those pools PG count. I thought it would increase PG count to 100 per OSD . I don't know why , so I have to turn off autoscale_mode for those pools. And let some pools stay as they are.

---
BTW. I restarted the 3 monitors one by one yestday , and all OSDs's RAW USE began to  decrease . I still don't know why.
CEPH just like a mystery
